### PR TITLE
TFAC-1071: Support authorization via auth cookie values

### PR DESCRIPTION
**Pull request by kmjennison**

### CODE REVIEW

**Comment by kmjennison** [comment]
I'm getting an unauthenticated user when testing on dev with what I believe is an Authorization header containing valid token values. Will debug.

File: lambda/src/firebase-authorizer/__tests__/firebase-authorizer.test.js [hunk 1]
```diff
@@ -249,6 +290,136 @@ describe('firebase-authorizer', () => {
...
+          {
+            Action: 'execute-api:Invoke',
+            Effect: 'Allow',
+            Resource: 'arn:execute-api:blah:blah',
+          },
+        ],
+      },
+      context: {
+        id: 'abc123xyz987',
+        email: 'meow@hogwarts.com',
```
Comment (jedtan) [inline lambda/src/firebase-authorizer/__tests__/firebase-authorizer.test.js:340]: nit, hogwarts.edu

File: lambda/src/firebase-authorizer/firebase-authorizer.js [hunk 2]
```diff
@@ -77,8 +78,30 @@ const checkUserAuthorization = async event => {
...
+    })
+    const user = {
+      uid: nfaUser.id,
+      email: nfaUser.email,
+      email_verified: nfaUser.emailVerified,
+      auth_time: 0, // NFA does not provide an auth time
+    }
+    return generatePolicy(user, true, event.methodArn)
+  } catch (e) {
+    // If not, don't throw. This is expected when provided with a Firebase
```
Comment (jedtan) [inline lambda/src/firebase-authorizer/firebase-authorizer.js:100]: Are there any other non - JSON parse errors we could get there? Might be useful to log if it's not an expected error type.

**Comment by jedtan** [comment]
Largely looks good to me, interested in hearing what the issue is.

File: lambda/src/firebase-authorizer/firebase-authorizer.js (hunk 2 above)
Comment (kmjennison) [reply lambda/src/firebase-authorizer/firebase-authorizer.js:100]: Good suggestion, will update.

**Comment by kmjennison** [comment]
It looks like the cookie secret is set or encrypted incorrectly. Will fix and re-test.

**Pull request merged**